src/client/Microsoft.Identity.Client/PublicClientApplication.cs

```diff
@@ -271,5 +272,26 @@ public bool IsBrokerAvailable()
                 username,
                 password);
         }
+
+        /// <summary>
+        /// Used to determine if the currently available broker is able to perform Proof-of-Possesion.
+        /// </summary>
+        /// <returns>Boolean indicating Proof-of-Possesion is supported</returns>
+        /// <exception cref="MsalClientException"></exception>
+        public bool IsProofOfPosessionSupportedByClient()
+        {
+            if (DesktopOsHelper.IsWin10OrServerEquivalent())
```

Comment:
you can combine these two ifs

Reply:
Since this is happening before an authentication call, there will be no logger available. Maybe if we refactor in the future to enable logging in this class.